src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -447,6 +460,85 @@ func (r *StatefulSetResource) queryRedpandaStatus(
 	return nil
 }
 
+// Temporarily using the status/ready endpoint until we have a specific one for restarting.
+func (r *StatefulSetResource) queryRedpandaUnderReplicatedPartition(
+	ctx context.Context, adminURL *url.URL,
+) error {
+	client := &http.Client{Timeout: adminAPITimeout}
+
+	// TODO right now we support TLS only on one listener so if external
+	// connectivity is enabled, TLS is enabled only on external listener. This
+	// will be fixed by https://github.com/redpanda-data/redpanda/issues/1084
+	if r.pandaCluster.AdminAPITLS() != nil &&
+		r.pandaCluster.AdminAPIExternal() == nil {
+		tlsConfig, err := r.adminTLSConfigProvider.GetTLSConfig(ctx, r)
+		if err != nil {
+			return err
+		}
+
+		client.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+		adminURL.Scheme = "https"
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, adminURL.String(), http.NoBody)
+	if err != nil {
+		return err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("getting broker metrics (%s): %w", adminURL.String(), errRedpandaNotReady)
+	}
+
+	var parser expfmt.TextParser
+	metrics, err := parser.TextToMetricFamilies(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	for name, metricFamily := range metrics {
+		if name != "vectorized_cluster_partition_under_replicated_replicas" {
```

Comment:
this is per node metric? Don't we have to look for metric for this particular node?

Reply:
I will ask core how to discover under replicated replicas

Reply:
To be honest I forget how I implemented this function. We query current POD. You can see that headless service with exact pod name is used

https://github.com/redpanda-data/redpanda/blob/c5bb116d0d217ad5fa2b3481dcb7dd62ec196075/src/go/k8s/pkg/resources/statefulset_update.go#L201

That means metrics comes from one POD.

Reply:
oh I see, thanks